bmc/redfish.go

```diff
@@ -416,11 +460,24 @@ func (r *RedfishBMC) GetStorages(systemUUID string) ([]Storage, error) {
 	return result, nil
 }
 
-func (r *RedfishBMC) getSystemByUUID(systemUUID string) (*redfish.ComputerSystem, error) {
+func (r *RedfishBMC) getSystemByUUID(ctx context.Context, systemUUID string) (*redfish.ComputerSystem, error) {
 	service := r.client.GetService()
-	systems, err := service.Systems()
+	var systems []*redfish.ComputerSystem
+	err := wait.PollUntilContextTimeout(
+		ctx,
+		r.options.ResourcePollingInterval,
+		r.options.ResourcePollingTimeout,
+		true,
+		func(ctx context.Context) (bool, error) {
+			if ctx.Err() != nil {
+				return false, ctx.Err()
+			}
```

Comment:
Not necessary I think, as `PollUntilContextTimeout` checks for context cancellation.